### PR TITLE
Check for null result of getFixVersion

### PR DESCRIPTION
**Pull request by davedevretro**
Issue https://github.com/jenkinsci/jira-plugin/issues/658

Check for null result of getFixVersions as per example on atlassian site:
https://developer.atlassian.com/server/jira/platform/creating-workflow-extensions/#step-2.-edit-the-code

Tested on my local Jenkins instance.
🚨 Please review the [guidelines for contributing](../blob/master/docs/CONTRIBUTING.md) to this repository.

- [x] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side) and not your master branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub 
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

**Comment by rantoniuk** [comment]
Thanks for the contribution! I see you have checked `Ensure you have provided tests` but I don't see any test file modified, did you miss adding it to the PR?

File: src/main/java/hudson/plugins/jira/JiraSession.java [hunk 1]
```diff
@@ -274,16 +274,23 @@ public void replaceFixVersion(String projectKey, String fromVersion, String toVe
...
 
                 Pattern fromVersionPattern = Pattern.compile(regEx);
-                for (Version currentVersion : issue.getFixVersions()) {
-                    Matcher versionToRemove = fromVersionPattern.matcher(currentVersion.getName());
-                    if (!versionToRemove.matches()) {
-                        newVersions.add(currentVersion);
+
+                Iterable<Version> versions = issue.getFixVersions();
+                if(versions != null) {
+                    for (Version currentVersion : versions) {
```
Comment (rantoniuk) [inline src/main/java/hudson/plugins/jira/JiraSession.java:280]: How about using [ListUtils.emptyIfNull()](https://commons.apache.org/proper/commons-collections/javadocs/api-4.2/org/apache/commons/collections4/ListUtils.html) instead of adding the `if` clause?

**Comment by davedevretro** [comment]
Sorry I thought this meant that I was confirming it was tested - unticked now.

**Review by rantoniuk — CHANGES_REQUESTED** [review]
Build fails because of format check failure.
You need to run `mvn spotless:apply` after all changes, before committing changes

File: src/main/java/hudson/plugins/jira/JiraSession.java (hunk 1 above)
Comment (rantoniuk) [reply src/main/java/hudson/plugins/jira/JiraSession.java:280]: Scratch that, we're stuck on version 3 for now, that doesn't have this.

Comment (rantoniuk) [reply src/main/java/hudson/plugins/jira/JiraSession.java:280]: Found another way using Java8 Optional:
```java
for (Version currentVersion : Optional.ofNullable(issue.getFixVersions()).orElse(Collections.emptyList())) {
```

Comment (davedevretro) [reply src/main/java/hudson/plugins/jira/JiraSession.java:280]: Is it starting to break readability for minimal gain at this point?

Comment (rantoniuk) [reply src/main/java/hudson/plugins/jira/JiraSession.java:280]: I think it looks cleaner than adding another nested `if` statement, especially when there is no else (maybe we should consider adding `LOGGING.fine()`) , but it's fine if you want to leave it as it is.

Would you be able to add a test for this? I think this should be pretty easy by mocking the Null response object.

**Comment by davedevretro** [comment]
updated formatting

File: src/main/java/hudson/plugins/jira/JiraSession.java (hunk 1 above)
Comment (davedevretro) [reply src/main/java/hudson/plugins/jira/JiraSession.java:280]: Couldn't work out how to get a unit test to work for this case - sorry.. :/
